### PR TITLE
chore: throw error when dll-plugin needs to generate multiple manifest files but the path config is incorrect.

### PR DESCRIPTION
**Pull request by cwj0417**
**What kind of change does this PR introduce?**

Output error information in specific dll-plugin usage scenarios. related to #18200.

**Did you add tests for your changes?**

Not yet, If necessary, I will add test cases after getting familiar with the test code.

**Does this PR introduce a breaking change?**

No

**What needs to be documented once your changes are merged?**

I will send another PR to document repo later.

### CODE REVIEW

File: lib/LibManifestPlugin.js [hunk 1]
```diff
@@ -53,6 +53,16 @@ class LibManifestPlugin {
...
+				let targetPaths = {};
+				compilation.chunks.forEach(chunk => {
+					const targetPath = compilation.getPath(this.options.path, {
+						chunk
+					});
+					if (Object.values(targetPaths).includes(targetPath)) {
+						callback(new Error("each chunk must have unique output path"));
+					}
+					targetPaths[chunk.id] = targetPath;
+				});
```
Comment (alexander-akait) [inline lib/LibManifestPlugin.js:65]: It is a bad solution for perf, just use:
```
// We throw an error when we have multiple chunks and no one file interlation
const needError = compilation.chunks.size > 1 && !this.options.path.includes("\[.+\]");
```

Comment (cwj0417) [reply lib/LibManifestPlugin.js]: thank you for your review, I have made some changes.

**Review by alexander-akait — COMMENTED** [review]
Let's add a test case, you can find them in `test/configCases`, we have errors.js file for tests errors

**Comment by cwj0417** [comment]
> Let's add a test case, you can find them in `test/configCases`, we have errors.js file for tests errors

Thanks for the guidance, I'll take a look

**Comment by cwj0417** [comment]
Hi, Alex, I have add test case, but there was new issue that `configCases/dll-plugin/0-create-dll-with-contenthash` case failed.

In this case, `LibManifestPlugin` will receive compilation with 2 chuncks. (and failed because is considered `needError` by previous code) 
In fact, the second loop will go into `chunk.canBeInitial()` branch and quit loop.
and we can't judge it before iteration, because the second chunk' s status changed in the first loop.

In this commit, if the running process does not go into the wrong process, each loop will just have additional overhead caused by array operation.
is this acceptable? I would like to ask your opinion regarding the above situation.

**Comment by alexander-akait** [comment]
Maybe we can use `const usedPaths = new Set();` and just throw an error if `usedPaths` already contains used path

**Comment by cwj0417** [comment]
> Maybe we can use `const usedPaths = new Set();` and just throw an error if `usedPaths` already contains used path

Thanks for your quick reply, I submitted the new code. (The test case has been run locally)

**Review by alexander-akait — APPROVED** [review]

**Pull request merged**